api-put-bucket.go

```diff
@@ -477,3 +479,95 @@ func (c Client) EnableVersioning(ctx context.Context, bucketName string) error {
 func (c Client) DisableVersioning(ctx context.Context, bucketName string) error {
 	return c.setVersioning(ctx, bucketName, versionDisableConfig, versionDisableConfigLen, versionDisableConfigMD5Sum, versionDisableConfigSHA256)
 }
+
+// RemoveBucketReplication removes a replication config on an existing bucket.
+func (c Client) RemoveBucketReplication(ctx context.Context, bucketName string) error {
+	return c.SetBucketReplication(ctx, bucketName, replication.Config{}, "")
+}
+
+// SetBucketReplication sets a replication config on an existing bucket.
+func (c Client) SetBucketReplication(ctx context.Context, bucketName string, cfg replication.Config, replicationArn string) error {
```

Comment:
We should start with `ReplicationOptions` as input @poornas for the API and this `opts.Validate()` would validate all the options..

Reply:
Not really required as of now - however I added an opts for future proofing.